modules/c++/sys/unittests/test_os.cpp

```diff
@@ -81,12 +81,70 @@ TEST_CASE(testForcefulMove)
     TEST_ASSERT( !os.exists(subdir1) );
 }
 
+TEST_CASE(testEnvVariables)
+{
+    sys::OS os;
+    const std::string testvar = "TESTVARIABLE";
+    const std::string testvalue = "TESTVALUE";
+    const std::string testvalue2 = "TESTVALUE2";
+
+    // Start by clearing the environment variable, if set.
+    os.unsetEnv(testvar);    
+    TEST_ASSERT(!os.isEnvSet(testvar));
+
+    // Check getEnv throws an sys::SystemException exception when trying unset var
+    try
```

Comment:
For code clarity, just use `TEST_EXCEPTION` here.  If philosophically you would like to check this precisely, add a new macro in TestCase.h that lets you do this.